macOS setup

zemtsov-artem opened the pull request.

zemtsov-artem committed:
Updated setup.py + added macos- requirements.file

vasily-v-ryabov commented on setup.py:
```diff
@@ -64,10 +64,22 @@ def setup_path(path = ""):
     packages = ["pywinauto", "pywinauto.tests", "pywinauto.controls", "pywinauto.linux"]
+elif sys.platform == 'darwin':
+    install_requires = [
+        'codecov', 
```
`codecov` is a requirement for running tests with coverage (for development only). `setup.py` must contain only production needed packages.

vasily-v-ryabov commented:
`dev-requirements.txt` contains all packages needed for development of pywinauto. This is not for end users. `setup.py` is for end user installation. There is new feature in `pip` to add `sys.platform` dependent packages into requirements file. Ideally we can modify `dev-requirements.txt` this way.

zemtsov-artem committed:
Removed unnecessary macos-req file.
Replaced dev packages from setup.py to dev-req file
Grouped dev packages by platfrom

zemtsov-artem committed:
Removed duplicated package

zemtsov-artem commented:
> `dev-requirements.txt` contains all packages needed for development of pywinauto. This is not for end users. `setup.py` is for end user installation. There is new feature in `pip` to add `sys.platform` dependent packages into requirements file. Ideally we can modify `dev-requirements.txt` this way.

Applied

vasily-v-ryabov commented on dev-requirements.txt:
```diff
@@ -1,10 +1,29 @@
-six
+# Win32 specific packages
+#
+pywin32                                   ;sys_platform == 'pywin32'
```
Maybe `sys_platform == 'win32'` ?

zemtsov-artem committed:
Applied comments

vasily-v-ryabov commented on dev-requirements.txt:
```diff
@@ -1,10 +1,30 @@
+#
+pywin32                                   ;sys_platform == 'win32'
+
+# Darwin specific packages
```
`Darwin` -> `macOS`, `Win32` -> `Windows`

zemtsov-artem committed:
Applied comments

zemtsov-artem committed:
Rebased onto latest code + fixed Codacy comments
